samples/hello-cube.html

```diff
@@ -0,0 +1,313 @@
+
+    const context = canvas.getContext('gpu');
+
+    const swapChainDescriptor = {
+        context: context,
+        format: "bgra8unorm"
+    };
+
+    const swapChain = device.createSwapChain(swapChainDescriptor);
+
+    /* Shader Module */
+    const shaderModuleDescriptor = { code: shader };
+    const shaderModule = device.createShaderModule(shaderModuleDescriptor);
+
+    const verticesBufferDescriptor = {
+        size: vertexArray.byteLength,
+        usage: GPUBufferUsage.VERTEX | GPUBufferUsage.TRANSFER_DST
+    };
+    verticesBuffer = device.createBuffer(verticesBufferDescriptor);
+    verticesBuffer.setSubData(0, vertexArray.buffer);
+
+    /* Input State */
+    const positionAttributeDescriptor = {
+        attributeIndex: 0,
+        offset: 0,
+        format: "float4"
+    };
+    const colorAttributeDescriptor = {
+        attributeIndex: 1,
+        offset: colorOffset,
+        format: "float4"
+    }
+    const vertexBufferDescriptor = {
+        stride: vertexSize,
+        stepMode: "vertex",
+        attributes: [positionAttributeDescriptor, colorAttributeDescriptor]
+    };
+    const inputStateDescriptor = {
+        indexFormat: "uint32",
+        vertexBuffers: { 0: vertexBufferDescriptor }
+    };
+
+    /* Bind Group Layout */
+    const transformBufferBindGroupLayoutBinding = {
+        binding: 0,
+        visibility: GPUShaderStageBit.VERTEX,
+        type: "uniform-buffer"
+    };
+
+    const uniformsBindGroupLayoutDescriptor = {
+        bindings: [transformBufferBindGroupLayoutBinding]
+    };
+    uniformsBindGroupLayout = device.createBindGroupLayout(uniformsBindGroupLayoutDescriptor);
+
+    const depthStateDescriptor = {
+        depthWriteEnabled: true,
+        depthCompare: "less"
+    };
+
+    /* Render Pipeline */
+    const vertexStageDescriptor = {
+        module: shaderModule,
+        entryPoint: "vertex_main"
+    };
+    const fragmentStageDescriptor = {
+        module: shaderModule,
+        entryPoint: "fragment_main"
+    };
+    const pipelineDescriptor = {
+        bindGroupLayouts: [uniformsBindGroupLayout],
+
+        vertexStage: vertexStageDescriptor,
+        fragmentStage: fragmentStageDescriptor,
+
+        primitiveTopology: "triangle-list",
+        depthStencilState: depthStateDescriptor,
+        inputState: inputStateDescriptor
+    };
+    pipeline = device.createRenderPipeline(pipelineDescriptor);
+
+    /* Renderpass Attachments */
+    let colorAttachment = {
+        // attachment is acquired in render loop.
+        loadOp: "clear",
+        storeOp: "store",
+        clearColor: { r: 0.5, g: 1.0, b: 1.0, a: 1.0 }
+    };
+
+    // GPUExtent3D
+    const depthSize = {
+        width: canvas.width,
+        height: canvas.height,
+        depth: 1
+    };
+
+    const depthTextureDescriptor = {
+        size: depthSize,
+        arrayLayerCount: 1,
+        mipLevelCount: 1,
+        sampleCount: 1,
+        dimension: "2d",
+        format: "d32-float-s8-uint",
```

Comment:
At a glance, this looks like a UUID. 🤮

Reply:
We should at least make them consistent with the other texture formats.

Reply:
I think I missed this one when updating the names.

Reply:
Oh, you forgot to include depth and stencil formats when you updated the list of texture formats in the IDL. I'll make a PR in a bit to add them.